doc/adr/0007-api-service.md

```diff
@@ -0,0 +1,106 @@
+# 7. API service
+
+Date: 2021-08-10
+
+## Status
+
+Proposed
+
+## Context
+
+As basic processes, `wego` interacts with the kubernetes api and the github api. Currently, this interaction occurs on the client side. So, for example, for 2 users to access the graphical interface, they both need to have at least `kubectl` and permissions on the` .kubeconfig` file to access the cluster. The proposal is to run this interaction logic directly in the cluster so that in the near future it allows the creation of more powerful features for `wego`. One of those features could be the exposition of the graphical interface in a simpler way. It would also allow people to write their own tools for wego.
+
+## Decision
+
+The proposal consists of extracting the API related to applications, which we will call `wego-api`, execute it within the cluster through using deployment and expose it through a service. The wego-api will be the only entity that will be able to interact directly with the Kubernetes API. In other words, the wego-api will expose the necessary options to interact with wego applications, abstracting the logic that interacts with the Kubernetes API.
+
+To illustrate, you could say that the `wego` command line could tell the wego-api "Add this application" and it, in turn, would tell Kubernetes "Apply these manifest files. "
+
+For security reasons, permissions will be implemented through RBAC so that the wego-api only has access to the resources it needs.
+
+With this proposal, the use of CLI options would still be required, however, it would be an intermediate step to eventually do
+a 100% migration into the cluster.
+
+Specifically, the proposed changes are as follows:
+
+### Create wego-api docker image and update authentication method.
+
+For the wego-api to run within the cluster, a docker image will need to exist that can run it.
+
+The part that needs to be run in this image is the HTTP server which is initialized in the `wego ui run` command.
+
+To continue communicating with the Kubernetes API, it will be required to change the authentication method to one that allows communication from within the cluster called in-cluster client configuration.
+
+Kubectl and flux will be added to this docker image as well.
+
+### Generate RBAC-based permissions to the wego-api and apply them when installing the GitOps runtime.
+
+Since the wego-api needs to access resources within the cluster, RBAC permissions will be applied so that there are no security gaps. In consideration that the functionality of the wego-api will be specified only in actions within the namespace where the GitOps runtime is executed, the Role and RoleBinding resources will be used. Since the scope of this type of resource is by namespace. 
```

Comment:
Will `wego gitops install` install this into the cluster?  If so, please add note about that.  If not, can you elaborate on how it is installed?

Reply:
There is a note about that at the end of this section `These resources will be applied when installing the GitOps engine.` https://github.com/weaveworks/weave-gitops/pull/590/files#diff-d9dd63abc835dd1b94cd996e0c5cc69d16c4126a4faccad2f6b1ee3934ecf4faR67.